### ISSUE TITLE
filter example: use a different model

### ISSUE THREAD

**Issue by drakkan** [nnsuite/nnstreamer#886]
I compiled the filter example and it works fine, then I tryed to use a different model, I converted the example model as explained here:

https://codelabs.developers.google.com/codelabs/tensorflow-for-poets-2-tflite/#2

but using the resulting model gives this error:

```
./nnstreamer_example_filter 
** Message: 19:06:28.671: start app..
** Message: 19:06:28.671: finished to load labels, total 5
nnapi error: unable to open library libneuralnetworks.so
** Message: 19:06:28.804: received start message
**
ERROR:/build/nnstreamer-3lS59X/nnstreamer-0.0.3/gst/tensor_filter/tensor_filter.c:893:gst_tensor_filter_configure_tensor: assertion failed: (0)
Aborted (core dumped)
```

the warning about libneuralnetworks.so is printed in the working case too, here is the output using the default model

```
/nnstreamer_example_filter 
** Message: 18:44:41.966: start app..
** Message: 18:44:41.971: finished to load labels, total 1001
nnapi error: unable to open library libneuralnetworks.so
** Message: 18:44:42.101: received start message
** Message: 18:44:42.502: receiving new data [1]
** Message: 18:44:42.868: receiving new data [2]
```

I'm using ubuntu ppa, what could be wrong here? thanks

**Comment by myungjoo** [comment]
@wooksong @again4you : I guess ```libneuralnetworks.so``` is being used by our own packaging of tensorflow-lite. Is that right? I think that warning message is harmless (tf-lite is going to use default backend if that is missing, but please confirm it.
2```.

@drakkan This error implies that the given input dimension of your model is not compatible with the hardcoded input dimension in the example code. You might need to write your own gstreamer pipeline for this.

@helloahn Please provided @drakkan an example code that is compatible with this model (I'm not sure if TFLITE's (1 - 224 - 224 - 3) is same as nnstreamer's 3:224:224 or 1:224:224:3 )

I'll add a commit that will show more "detailed info" with such errors.

**Comment by myungjoo** [comment]
@draken It is because ```nnstreamer_example_filter``` is configured to use 3:224:224/uint8 and your model accepts 3:224:224/float32. You can either:

1. Add ```tensor_transform``` with ```typecast``` mode (convert to float32) before ```tensor_filter```.
2. Convert the model to accept uint8 instead of float32

Note that ```tensor_converter``` always generate tensors with uint8 type for RGB video inputs.

**Comment by helloahn** [comment]
@drakkan 
it's about the datatype of the pipeline.
The type of data flow in gst pipeline is `uint8_t`.
However, your model uses `float` as input/output data type because the [pbfile](https://github.com/googlecodelabs/tensorflow-for-poets-2/blob/master/android/tfmobile/assets/graph.pb) has been frozen from float model.
```bash
$ bazel-bin/tensorflow/tools/graph_transforms/summarize_graph --in_graph=graph.pb
Found 1 possible inputs: (name=input, type=float(1), shape=[1,224,224,3]) 
No variables spotted.
Found 1 possible outputs: (name=MobilenetV1/Predictions/Reshape_1, op=Reshape) 
Found 1343080 (1.34M) const parameters, 0 (0) variable parameters, and 0 control_edges
Op types used: 166 Const, 138 Identity, 81 Mul, 54 Add, 27 Relu6, 27 Rsqrt, 27 Sub, 15 Conv2D, 13 DepthwiseConv2dNative, 2 Reshape, 1 AvgPool, 1 BiasAdd, 1 Placeholder, 1 Softmax, 1 Squeeze
To use with tensorflow/tools/benchmark:benchmark_model try these arguments:
bazel run tensorflow/tools/benchmark:benchmark_model -- --graph=graph.pb --show_flops --input_layer=input
```

To apply `float` datatype in gst pipeline, you have to do `typecast` before send data to `tensor_filter`.
We use `tensor_transform` for the typecast usually.

Below code will help you to understand.

https://github.com/nnsuite/nnstreamer/blob/fda99919133baab5ed59550cde044b7f57a8077c/nnstreamer_example/example_object_detection/nnstreamer_example_object_detection.cc#L677

**Comment by myungjoo** [comment]
After #889, with environement variable ```GST_DEBUG=1```, you will be able to clearly see what's wrong:
```
$ GST_DEBUG=1 ./nnstreamer_example_filter
Setting pipeline to PAUSED ...
Pipeline is live and does not need PREROLL ...
Setting pipeline to PLAYING ...
New clock: GstSystemClock
0:00:00.220844133  1885      0x181c2d0 ERROR          tensor_filter tensor_filter.c:953:gst_tensor_filter_configure_tensor:<tensorfilter0> The input tensor is not compatible.
 0 : float64 [3:300:300:1]  | float32 [3:300:300:1]

0:00:00.221013044  1885      0x181c2d0 ERROR          tensor_filter tensor_filter.c:953:gst_tensor_filter_configure_tensor:<tensorfilter0> The input tensor is not compatible.
 0 : uint8 [3:224:224:1]  | float32 [3:224:224:1]

0:00:00.221277027  1885      0x181c2d0 ERROR          tensor_filter tensor_filter.c:953:gst_tensor_filter_configure_tensor:<tensorfilter0> The input tensor is not compatible.
 0 : uint8 [3:224:224:1]  | float32 [3:224:224:1]

0:00:00.427684782  1885      0x181c2d0 ERROR          tensor_filter tensor_filter.c:953:gst_tensor_filter_configure_tensor:<tensorfilter0> The input tensor is not compatible.
 0 : uint8 [3:224:224:1]  | float32 [3:224:224:1]

0:00:00.443413796  1885      0x181c2d0 ERROR          tensor_filter tensor_filter.c:953:gst_tensor_filter_configure_tensor:<tensorfilter0> The input tensor is not compatible.
 0 : uint8 [3:224:224:1]  | float32 [3:224:224:1]

Redistribute latency...
```

**Comment by myungjoo** [comment]
If you want to keep using your tf-lite converted model (float32), you can modify the example to transform the type to float32 by:
```
NNStreamer/nnstreamer_example/example_filter$ git diff
diff --git a/nnstreamer_example/example_filter/nnstreamer_example_filter.c b/nnstreamer_example/example_filter/nnstreamer_example_filter.c
index ab82981..c964e00 100644
--- a/nnstreamer_example/example_filter/nnstreamer_example_filter.c
+++ b/nnstreamer_example/example_filter/nnstreamer_example_filter.c
@@ -444,6 +444,7 @@ main (int argc, char **argv)
       "t_raw. ! queue ! textoverlay name=tensor_res font-desc=Sans,24 ! "
       "videoconvert ! ximagesink name=img_tensor "
       "t_raw. ! queue leaky=2 max-size-buffers=2 ! videoscale ! video/x-raw,width=%d,height=%d ! tensor_converter ! "
+      "tensor_transform mode=arithmetic option=typecast:float32 ! "
       "tensor_filter framework=tensorflow-lite model=%s ! "
       "tensor_sink name=tensor_sink",
       width, height, g_app.tflite_info.model_path);
NNStreamer/nnstreamer_example/example_filter$ 
```
